pkg/kclient/deployments.go

```diff
@@ -115,6 +116,40 @@ func (c *Client) ListDeployments(selector string) (*appsv1.DeploymentList, error
 	})
 }
 
+func (c *Client) WaitForPodNotRunning(name string) error {
+	watch, err := c.KubeClient.CoreV1().Pods(c.Namespace).Watch(context.TODO(), metav1.ListOptions{FieldSelector: "metadata.name=" + name})
+	if err != nil {
+		return err
+	}
+	defer watch.Stop()
+
+	if _, err = c.KubeClient.CoreV1().Pods(c.Namespace).Get(context.TODO(), name, metav1.GetOptions{}); kerrors.IsNotFound(err) {
+		fmt.Printf("\npod not found\n")
+		return nil
+	}
+
+	for {
+		select {
+		case <-time.After(time.Minute):
+			return errors.New("timeout")
+
+		case val, ok := <-watch.ResultChan():
+			if !ok {
+				return errors.New("error getting value from resultchan")
+			}
+			if pod, ok := val.Object.(*corev1.Pod); ok {
+				for _, cond := range pod.Status.Conditions {
+					if cond.Type == "Ready" {
```

Comment:
Shouldn't we wait for the pod status to be `terminating` or something similar as the name of the function is `WaitForPodNotRunning`?

Reply:
I changed the function name to `WaitForPodNotReady`, as we really want to wait for the Ready condition to be False